datafusion/expr/src/expr_fn.rs

```diff
@@ -737,8 +737,9 @@ scalar_expr!(
     "converts the Unicode code point to a UTF8 character"
 );
 scalar_expr!(Digest, digest, input algorithm, "compute the binary hash of `input`, using the `algorithm`");
-scalar_expr!(Encode, encode, input encoding, "encode the `input`, using the `encoding`. encoding can be base64 or hex");
-scalar_expr!(Decode, decode, input encoding, "decode the`input`, using the `encoding`. encoding can be base64 or hex");
+// TODO make a variant of Expr that can invoke a function by name
```

Comment:
Supporting these functions will take some additional work, but I think it is doable